solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java

```diff
@@ -185,7 +184,9 @@ private DocCollection fetchCollectionState(SolrClient client, String collection)
     SimpleOrderedMap<?> cluster =
         submitClusterStateRequest(client, collection, ClusterStateRequestType.FETCH_COLLECTION);
 
-    var collStateMap = (Map<String, Object>) cluster.findRecursive("collections", collection);
+    var collState = (Map<String, Object>) cluster.findRecursive("collections");
```

Comment:
so findRecursive can only resolve collections and not Map?  Seems worth a small enhancement to do so but I understand if you don't bother or defer.

Reply:
Made the small enhancement to NamedList and added a test to make sure it works. Lmk what you think.